### PR TITLE
Added listeners to virtual-router under VirtualService CRD

### PR DESCRIPTION
**Pull request by kiranmeduri**
*Issue #, if available:*
#37 

*Description of changes:*
[VirtualRouter](https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_VirtualRouterSpec.html) allows listener to be configured hence this  update to CRD. 

Also includes
- modified color.yaml to be a template with mesh-name, namespace and images injectable for local dev.
- updated aws sdk

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controller/virtualservice.go [hunk 1]
```diff
@@ -288,6 +296,30 @@ func vserviceNeedsUpdate(desired *appmeshv1beta1.VirtualService, target *aws.Vir
...
+	if desired.Name != target.Name() {
+		return true
+	}
+
+	if len(desired.Listeners) != len(target.Data.Spec.Listeners) {
+		return true
+	}
+
+	if len(desired.Listeners) > 0 {
+		//there should be only one listener
```
Comment (mogren) [inline pkg/controller/virtualservice.go]: Somewhat tricky logic to read, but it makes sense. I was just considering the options:
```
+------------------------+---------------------------------+-------------------+
| len(desired.Listeners) | len(target.Data.Spec.Listeners) | routerNeedsUpdate |
+------------------------+---------------------------------+-------------------+
|           0            |               0                 |       false       |
|                        |                                 |                   |
|           1            |               0                 |       true        |
|                        |                                 |                   |
|           0            |               1                 |       true        |
|                        |                                 |                   |
|           1            |               1                 |       true        |
|                        |                                 |                   |
|           2            |               1                 |       true        |
+------------------------+---------------------------------+-------------------+
```
I assume that `target.Data.Spec.Listeners` contains at most `1` listener?

File: pkg/controller/virtualservice_test.go [hunk 2]
```diff
@@ -126,13 +137,99 @@ func TestVServiceNeedsUpdate(t *testing.T) {
...
+
+	var tests = []struct {
+		name        string
+		spec        *appmeshv1beta1.VirtualRouter
+		aws         *aws.VirtualRouter
+		needsUpdate bool
+	}{
+		{"virtual-routers are the same", defaultRouter, defaultResult, false},
+		{"virtual-routers has different router name", defaultRouter, resultDifferentRouterName, true},
+		{"virtual-routers has different listener", defaultRouter, resultDifferentRouterListener, true},
```
Comment (mogren) [inline pkg/controller/virtualservice_test.go]: Nice tests

File: examples/color.yaml.template [hunk 3]
```diff
@@ -4,119 +4,129 @@ kind: Namespace
...
+          protocol: http
   routes:
     - name: color-route
       http:
         match:
           prefix: /
         action:
           weightedTargets:
-            - virtualNodeName: colorteller.appmesh-demo
+            - virtualNodeName: colorteller
```
Comment (jqmichael) [inline examples/color.yaml.template]: Would the original version still work?

Comment (kiranmeduri) [reply examples/color.yaml.template]: yes it does

File: pkg/controller/virtualservice.go (hunk 1 above)
Comment (kiranmeduri) [reply pkg/controller/virtualservice.go]: yeah I can change this to a loop and assume order is preserved or be defensive and error if > 1. either ways it is still tricky :)

File: examples/color.yaml.template (hunk 3 above)
Comment (jqmichael) [reply examples/color.yaml.template]: OK. Thanks for the confirmation.

**Comment by nckturner** [comment]
Ha, started this review yesterday and didn't check to see it was merged :D

**Comment by mogren** [comment]
Sorry @nckturner, didn't want to bother you with it this week 😄

**Pull request merged**